### PR TITLE
TracerProvider renamings

### PR DESCRIPTION
**Pull request by cijothomas**
API usage:
```
var tracerProvider = TracerProvider.Default;
var tracer = tracerProvider.GetTracer("MyCompany.MyProduct.MyWebServer");
tracer.StartSpan("myspan");
```

The above results in noops - no spans are created or exporter.

Enable SDK as below:

```
using var tracerProvider = OpenTelemetrySdk.CreateTracerProvider(
                (builder) => builder.AddActivitySource("MyCompany.MyProduct.MyWebServer"))
```

Now all spans from the configured source get exported :)

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/Trace/TracerProviderSdk.cs [hunk 1]
```diff
@@ -17,171 +17,49 @@
...
 namespace OpenTelemetry.Trace
 {
-    public class TracerProviderSdk : TracerProvider, IDisposable
+    internal class TracerProviderSdk : TracerProvider, IDisposable
     {
-        private readonly List<object> instrumentations = new List<object>();
-        private Resource resource;
-        private ActivityProcessor activityProcessor;
-        private ActivityListener listener;
+        internal readonly List<object> Instrumentations = new List<object>();
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Trace/TracerProviderSdk.cs:27]: You could make these public, the class itself is internal.

File: src/OpenTelemetry.Api/Trace/TracerProvider.cs [hunk 2]
```diff
@@ -14,22 +14,40 @@
...
-    /// TracerProvider is the entry point of the OTel API. It provides access to Tracers.
+    /// TracerProvider is the entry point of the OpenTelemetry API. It provides access to Tracers.
     /// </summary>
-    public abstract class TracerProvider
+    public class TracerProvider : IDisposable
     {
+        /// <summary>
+        /// Gets the default Tracer.
+        /// </summary>
+        public static TracerProvider Default = new TracerProvider();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/TracerProvider.cs:30]: It's weird to have a public static field. Probably should be:
```csharp
 public static TracerProvider Default { get; } = new TracerProvider();
```
If you meant it to be settable, what's the use case for that?

File: src/OpenTelemetry.Api/Trace/TracerProvider.cs [hunk 3]
```diff
@@ -14,22 +14,40 @@
...
+        }
+
         /// <summary>
         /// Gets a tracer with given name and version.
         /// </summary>
         /// <param name="name">Name identifying the instrumentation library.</param>
         /// <param name="version">Version of the instrumentation library.</param>
         /// <returns>Tracer instance.</returns>
-        public static Tracer GetTracer(string name, string version = null)
+        public Tracer GetTracer(string name, string version = null)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/TracerProvider.cs:50]: Each Tracer news up an ActivitySource so users will also have to add that to their list of captured sources when they enable SDK? That's a little different from what we have today. Today if you do TracerFactoryBase.Default.GetTracer (something like that) and emit spans through that Tracer, they are automatically included. Not necessarily a problem, but I think it's a big change we should consider.

File: src/OpenTelemetry.Api/Trace/TracerProvider.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/TracerProvider.cs:30]: Its not meant to be settable (atleast in the current design). Picking your suggestion! thanks.

File: src/OpenTelemetry.Api/Trace/TracerProvider.cs (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/TracerProvider.cs:50]: Yes its a bit less user friendly, as one has to remember to include the ActivitySource to the list of sources to listen.
We need to find alternate and better ways to this problem. Maybe change listening model to be "listen to all, except those are explicitly excluded" or "listen to all sources starting with MyCompany.MyProduct.*" etc.
https://github.com/open-telemetry/opentelemetry-dotnet/issues/736

**Pull request merged**